From 1169d34a65d46b1e4b40a77ee221cb0166fd244a Mon Sep 17 00:00:00 2001
From: Teruyuki Imai <t-imai@cp.jp.nec.com>
Date: Thu, 12 Oct 2017 14:19:06 +0900
Subject: [PATCH] task refs #104

Declare a global variable in libvepseudo in header as extern.
Without extern, the variable in BSS is created in a shared library
instead of referring to the symbol in libvepseudo.
---
 lib/libvepseudo/amm/vemva_mgmt.h | 5 +++--
 lib/libvepseudo/loader/loader.h  | 6 ++++--
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/lib/libvepseudo/amm/vemva_mgmt.h b/lib/libvepseudo/amm/vemva_mgmt.h
index 37e5083..fe6d378 100644
--- a/lib/libvepseudo/amm/vemva_mgmt.h
+++ b/lib/libvepseudo/amm/vemva_mgmt.h
@@ -89,7 +89,7 @@
 #define MARK_VESHM		(((uint8_t)1)<<4)
 #define MARK_FILE		(((uint8_t)1)<<5)
 
-uint64_t default_page_size;
+extern uint64_t default_page_size;
 /**
  * @brief Structure to store ve page info
  */
@@ -111,7 +111,8 @@ struct vemva_header {
 	uint64_t ve_heap;		/*!< ve heap section address */
 	uint64_t ve_stack;		/*!< ve stack section address */
 	struct list_head vemva_list;	/*!< list head for vemva list*/
-} vemva_header;
+};
+extern struct vemva_header vemva_header;
 
 /**
  * @brief Data Structure for VEMVA List
diff --git a/lib/libvepseudo/loader/loader.h b/lib/libvepseudo/loader/loader.h
index 9aa2033..b2e8b78 100644
--- a/lib/libvepseudo/loader/loader.h
+++ b/lib/libvepseudo/loader/loader.h
@@ -80,7 +80,8 @@ struct ve_address_space_info {
 	uint64_t stack_bottom;		/*!< Stack bottom of a VE Process */
 	uint64_t stack_pointer_aligned;	/*!< Page size aligned actual stack
 					 * pointer of a VE process. */
-} ve_info;
+};
+extern struct ve_address_space_info ve_info;
 
 /**
 * @brief Contain information for executable and library to load.
@@ -134,7 +135,8 @@ struct ve_load_data {
 	char *ve_interp_map;		/*!< VE Address at which interp has
 					 * been mapped*/
 	uint64_t pagesize;
-} load_elf;
+};
+extern struct ve_load_data load_elf;
 
 void ve_set_stack_limit(void);
 int ve_address_space_reserve(uint64_t);
-- 
1.8.3.1

